File: AirportChallenge/spec/Weather_spec.rb

require 'weather'
require 'airport'

describe Weather do

  let(:airport) {double :airport}

  it 'stormy is true' do
    allow(airport).to receive(:stormy?) {true}
    expect(airport.stormy?).to eq(true) | eq(false)
  end

  it 'stormy is false' do
    allow(airport).to receive(:stormy?) {false}
    expect(airport.stormy?).to eq(true) | eq(false)
  end
end

File: JavaScriptAirport/spec/weatherSpec.js
'use strict'

describe('Weather', function() {
  var weather;

  beforeEach(function() {
    weather = new Weather();
  });
  it('is stormy sometimes', function() {
    spyOn(Math,'random').and.returnValue(1);
    expect(weather.isStormy()).toBeTruthy();
  });
  it('is not stormy other times', function() {
    spyOn(Math,'random').and.returnValue(0);
    expect(weather.isStormy()).toBeFalsy();
  });
});
